client/src/main/java/io/pravega/client/stream/impl/PositionImpl.java

```diff
@@ -66,5 +76,53 @@ public Long getOffsetForOwnedSegment(Segment segmentId) {
     public PositionImpl asImpl() {
         return this;
     }
+    
+    @Override
+    public String toString() {
+        return ToStringUtils.mapToString(ownedSegments);
+    }
+
+    private static class PositionBuilder implements ObjectBuilder<PositionImpl> {
+    }
+    
+    public static class EventPointerSerializer extends VersionedSerializer.WithBuilder<PositionImpl, PositionBuilder> {
+        @Override
+        protected PositionBuilder newBuilder() {
+            return builder();
+        }
+
+        @Override
+        protected byte getWriteVersion() {
+            return 0;
+        }
+
+        @Override
+        protected void declareVersions() {
+            version(0).revision(0, this::write00, this::read00);
+        }
+
+        private void read00(RevisionDataInput revisionDataInput, PositionBuilder builder) throws IOException {
+            Map<Segment, Long> map = revisionDataInput.readMap(in -> Segment.fromScopedName(in.readUTF()), in -> in.readCompactLong());
```

Comment:
`in -> in.readCompactLong()` -> `RevisionDataInput::readCompactLong`

Reply:
This can be done in multiple places. It's just a personal preference; there's nothing wrong with the code you wrote.